### ISSUE TITLE
invalid argument 2: hex number with leading zero digits

### ISSUE THREAD

**Issue by pata-eth** [ledgerwatch/erigon#7548]
Hi, folks. When running `eth_call` with overrides below I get 

```bash
invalid argument 2: hex number with leading zero digits
```

but I'm able to successfully run it if I fork the live network with ganache. The tx is a `transferFrom` with overrides.

```bash
curl -X POST <Erigon RPC URL> -H "Content-Type: application/json" --data '{"jsonrpc": "2.0", "method": "eth_call", "params": [{"from":"0xb634316E06cC0B358437CbadD4dC94F1D3a92B3b", "to":"0x3506424F91fD33084466F402d5D97f05F8e3b4AF", "data":"0x23b872dd000000000000000000000000cdb90e5a0d06f35a71e96467e46ae1822510071e000000000000000000000000b634316e06cc0b358437cbadd4dc94f1d3a92b3b0000000000000000000000000000000000000000000000000de0b6b3a7640000"},"latest", {"0x3506424F91fD33084466F402d5D97f05F8e3b4AF": {"stateDiff":{"0x9a8b3209a5004a31d98bbf85891c8f12d340ca14b5449dec7f712548fee0da70":"0x00000000000000000000000000000000000000007fffffffffffffffffffffff","0x6a5dadead0dffc0ee6627048a9d1cdfa1037189dd1456007ef92bbcda4a84cef":"0x00000000000000000000000000000000000000007fffffffffffffffffffffff"}}}], "id":1}' 
```

Any idea why I'm getting that error when using the live network? Many thanks

**Comment by lelerukjaymoh** [comment]
Facing the same issue, @pata-eth did you get it to work?

**Comment by pata-eth** [comment]
> Facing the same issue, @pata-eth did you get it to work?

I did not. The fork sufficed for my use case but definitely something that should be addressed when running live networks imo.

**Comment by calmbeing** [comment]
@pata-eth Try not pass number arguments with leading zero digits. Like in your case, `0x00000000000000000000000000000000000000007fffffffffffffffffffffff` => `0x7fffffffffffffffffffffff`

**Comment by pata-eth** [comment]
hey! I just noticed the reply. Thank you! I actually stumbled upon the same question over a year later and here I am lol

Would it be possible to replicate the behavior in other clients where all overrides are padded to be 32bytes? In many cases, we don't know the type of the overriden variable (in order to know when to remove the leading zeroes). For example, in my workflow I run a `trace_call` to identify storage changes needed before I run a simulation (an eth_call). I'm not really interested in their types; just that these are correct for my simulation to succeed.

Thank you
